scripts/irods/test/test_pam_password.py

```diff
@@ -0,0 +1,225 @@
+
+        finally:
+            self.auth_session.environment_file_contents = auth_session_env_backup
+            for filename in [chain_pem_path, server_key_path, dhparams_pem_path]:
+                if os.path.exists(filename):
+                    os.unlink(filename)
+
+            IrodsController().restart()
+
+
+class test_iadmin_set_pam_password_config(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        self.admin = session.mkuser_and_return_session('rodsadmin', 'otherrods', 'rods', lib.get_hostname())
+
+        self.default_setup = {
+            "no_extend": "",
+            "password_length": "",
+            "password_max_time": "",
+            "password_min_time": ""
+        }
+
+        _, out, _ = self.admin.assert_icommand(['iadmin', 'get_pam_password_config'], 'STDOUT', 'password_length')
+        self.original_configs = json.loads(out)
+
+        # Set the pam_password configuration values to the defaults.
+        for option_name in self.default_setup:
+            self.admin.assert_icommand(
+                ['iadmin', 'set_pam_password_config', option_name, self.default_setup[option_name]])
+
+    @classmethod
+    def tearDownClass(self):
+        # Put everything back to the original configuration settings.
+        for option_name in self.original_configs:
+            self.admin.assert_icommand(
+                ['iadmin', 'set_pam_password_config', option_name, self.original_configs[option_name]])
+
+        with session.make_session_for_existing_admin() as admin_session:
+            self.admin.__exit__()
+            admin_session.assert_icommand(['iadmin', 'rmuser', self.admin.username])
+
+    def test_no_parameters(self):
+        self.admin.assert_icommand(
+            ['iadmin', 'set_pam_password_config'], 'STDERR', 'Invalid pam_password configuration option')
+
+        # Assert that nothing changed due to the error.
+        expected_output = self.default_setup
+        self.admin.assert_icommand(
+            ['iadmin', 'get_pam_password_config'], 'STDOUT', json.dumps(expected_output, indent=4))
+
+    def test_invalid_option_name(self):
+        self.admin.assert_icommand(
+            ['iadmin', 'set_pam_password_config', 'invalid'], 'STDERR', 'Invalid pam_password configuration option')
+
+        # Assert that nothing changed due to the error.
+        expected_output = self.default_setup
+        self.admin.assert_icommand(
+            ['iadmin', 'get_pam_password_config'], 'STDOUT', json.dumps(expected_output, indent=4))
+
+    def test_empty_value(self):
+        option_name = 'password_length'
+        option_value = 500
+        self.admin.assert_icommand(['iadmin', 'set_pam_password_config', option_name, str(option_value)])
+
+        # Assert that the value has been updated to the expected new value.
+        expected_output = self.default_setup.copy()
+        expected_output['password_length'] = str(option_value)
```

Comment:
You could replace the string literal with `option_name`?

Up to you.

Reply:
Yep, will do

Reply:
Done